### DISCUSSION TITLE
Customize report

### DISCUSSION THREAD

**Discussion by ericfabiszewski** [artilleryio/artillery#1364] [Artillery]
I am new to artillery but is there any way to customize the report, specifically the graph to not show the users created portions at all and just the requests information?

**Comment by hassy** [comment]
Not at the moment @ericfabiszewski. Is VU information not relevant for your use case at all?

**Reply by ericfabiszewski** [reply]
Not really, just wanted to be able to remove things that manager does not really want to focus on

**Comment by GeekyDeaks** [comment]
I did something similar by creating a modified report based on https://github.com/artilleryio/artillery/blob/master/lib/cmds/report.js

```
e.g.
const fs = require('fs/promises');
const path = require('path');
const _ = require('lodash')

async function report(args = {}) {
    // taken from https://github.com/artilleryio/artillery/blob/master/lib/cmds/report.js

    const output = args.output || args.file + '.html'; // TODO: path.resolve
    const data = JSON.parse(await fs.readFile(args.file, 'utf-8'));
    data.intermediate.forEach((o) => delete o.latencies); 
    const templateFn = path.join(
      path.dirname(__filename),
      'node_modules', 'artillery', 'lib', 'report', 'index.html.ejs'
    )
    const template = await fs.readFile(templateFn, 'utf-8')
    const compiledTemplate = _.template(template)
    const html = compiledTemplate({ report: JSON.stringify(data, null, 2) })
    await fs.writeFile(output, html, { encoding: 'utf-8', flag: 'w' })
    console.log('Report generated: %s', output)
}
```

You can then remove the unwanted metrics from `data.intermediate` or maybe copy and modify `index.html.ejs`

**Reply by ericfabiszewski** [reply]
Thank you for your response!  How would i actually call this in the script?  Would this be in a custom JS file?

**Reply by GeekyDeaks** [reply]
Yes, it would be in a custom JS file, which you would run after artillery had finished and generated the .json output file.  The above function is one I use in a simple .js library file to help programmatically launch artillery and then transform and convert the reports.  It might not be super clear in the code, but actually the process of converting the report is very simple and uses the template function in lodash: https://lodash.com/docs/4.17.15#template
